install/first_google_boot.sh

```diff
@@ -224,11 +224,22 @@ function extract_spinnaker_kube_credentials() {
   chown -R spinnaker:spinnaker $(dirname $config_path)
 
   local kube_cluster=$(get_instance_metadata_attribute "kube_cluster")
+  local kube_project=$(get_instance_metadata_attribute "kube_project")
+  local kube_zone=$(get_instance_metadata_attribute "kube_zone")
+
+  if [ -z "$kube_project" ]; then
+    kube_project=$MY_PROJECT
+  fi
+
+  if [ -z "$kube_zone" ]; then
+    kube_zone=$MY_ZONE
+  fi
+
   if [ -n "$kube_cluster" ]; then
```

Comment:
what happens when you leave something out, misspell it or cluster isnt there? I think this will prevent the startup script from running.  I guess that is ok, except you should run "first_google_boot" after fixing the problem and, if the problem is in the metadata, you have to trash the VM and start over again. So I think as you add more parameters here, error recovery becomes more significant to address (and more complex).

With the spinnaker-local, if you mess it up, it will still extract out to disk, spinnaker will fail to start (or configure incorrectly) but you could edit the file and restart spinnaker using normal/standard practices.

Can you think through error scenarios and how to recover from them? I dont know how important that is. I assume that in practice use of these fields would be scripted so error rates might not be that high. But it would be easy to leave out a project/zone or add a wrong one when supplying script parameters. And C2D doesnt provide the means to check it. I think that will cause the script to abort early in a non-recoverable way. Although you could fix it manually, you'd never be able to complete the other parts of the script. That feels bad.